spec/scene-perception.html

```diff
@@ -1194,6 +1194,168 @@
+            var initConfig = {
+              useOpenCVCoordinateSystem: true,
+              colorCaptureSize: {'width':320, 'height':240},
+              depthCaptureSize: {'width':320, 'height':240},
+              captureFramerate: 60
+            };
+            sp.init(initConfig).then(function() {
+              // Set the state of buttons.
+              resetButtonState(false);
+
+              // Other initialization work can be done here.
+              console.log('init succeeds');
+            }, function(e) {console.log(e);});
+          };
+
+          startButton.onclick = function(e) {
+            sp.start().then(function() {
+              startButton.disabled = true;
+              stopButton.disabled = false;
+              console.log('SP started successfully');
+            }, function(e) {console.log(e);});
+          };
+
+          stopButton.onclick = function(e) {
+            sp.stop().then(function() {
+              console.log('SP stops working.');
+              startButton.disabled = false;
+              stopButton.disabled = true;
+            }, function(e) {console.log(e);});
+          };
+
+          destroyButton.onclick = function(e) {
+            sp.destroy().then(function() {
+              console.log('stop succeeds');
+              resetButtonState(true);
+            }, function(e) {console.log(e);});
+          };
+        </pre>
+      </section>
+      <section>
+        <h3>
+          Handle events of SP module
+        </h3>
+        <pre class='example highlight'>
+          var qualityElement = document.getElementById('quality');
+          var accuracyElement = document.getElementById('accuracy');
+
+          sp.onchecking = function(e) {
+            var quality = e.data.quality;
+            qualityElement.innerHTML = 'Quality: ' + quality.toFixed(2);
+          };
+
+          // Please refer the SampleProcessedEvent inferface for event data.
+          sp.onsampleprocessed = function(e) {
+            accuracyElement.innerHTML = 'Accuracy: ' + e.data.accuracy;
+            qualityElement.innerHTML = 'Quality: ' + e.data.quality.toFixed(2);
+
+            sp.getSample().then(function(sample) {
+              // The sample object contains color image and depth image.
+              // The image data structure, which could possibly looks like this:
+              // sample.color = {
+              //                  width: imageWidth,
+              //                  height: imageHeight,
+              //                  data: imageDataArrayBuffer
+              //                };
+              // sample.depth = {
+              //                  width: imageWidth,
+              //                  height: imageHeight,
+              //                  data: imageDataArrayBuffer
+              //                };
+              // The imageDataArrayBuffer is an ArrayBuffer with type of Uint32Array, and the byte
```

Comment:
For color image, the data should be a Uint8Array. For depth image, the data should be a Uint16Array.